cmd/multiclusterapp.go

```diff
@@ -611,32 +734,25 @@ func multiClusterAppTemplateInstall(ctx *cli.Context) error {
 		return err
 	}
 
-	var (
-		timewait  int
-		installed bool
-	)
-	timeout := ctx.Int("timeout")
-	for !installed {
-		if timewait*2 >= timeout {
+	fmt.Printf("Installing multi-cluster app %q...\n", appName)
+
+	return waitUntilMultiClusterAppActive(c, madeApp.ID, ctx.Int("timeout"))
+}
+
+func waitUntilMultiClusterAppActive(c *cliclient.MasterClient, appID string, timeout int) error {
+	var timeWait int
+	for {
+		if timeWait*2 >= timeout {
 			return errors.New("timed out waiting for app to be active, the app could still be installing. Run 'rancher multiclusterapps' to verify")
 		}
-		timewait++
+		timeWait++
```

Comment:
Can we do `context.Timeout` and `time.Ticker` here? It would be better than we are using an endless loop here since we are changing this.

Reply:
Updated